include/JSONFormatter.hpp

```diff
@@ -0,0 +1,33 @@
+/*
+*   JSONFormatter.hpp
+*   ----------------------
+*   Created on: 2023/10/20
+*   Author: Lankow
+*/
+#ifndef JSON_FORMATTER_HPP
+#define JSON_FORMATTER_HPP
+#include "DataProvider.hpp"
+
+class JSONFormatter{
+  private: 
+    DataProvider* m_dataProvider;
+    String buildHumidityJson();
```

Comment:
Czym jest `String`? To z jakieś zewnetrznej libki? W c++ jest `std::string` albo `string` jeśli załadujesz sobie namespace `std`

Reply:
"They are just different implementations of the same principle. std::string is the most commonly used version, as it is part of the official C++ standard. However, on AVR Arduinos, you cannot use the C++ Standard Template Library (STL), so you cannot use std::string. That's why Arduino created its own String class, and this is the most widely used version in the Arduino ecosystem.

While both string classes use dynamic memory, most implementations of std::string have Small String Optimizations (SSO), where small strings are allocated on the stack instead of the heap. This can slightly reduce the amount of fragmentation caused by the strings.

The best string implementation to use on microcontrollers with limited RAM is null-terminated character arrays, allocated on the stack. The stack data structure ensures that memory is deallocated in the reverse order of the allocations, which makes fragmentation impossible. These strings are often referred to as "c-strings"."